platform/pulp/platform/models/catalog.py

```diff
@@ -0,0 +1,33 @@
+"""
+Django models for artifact catalogs.
+"""
+from django.db import models
+
+from pulp.platform.models import Model, Artifact, RepositoryImporter
+
+
+class DownloadCatalog(Model):
+    """
+    A catalog that tracks where artifacts originated from.
+
+    Fields:
+
+    :cvar url: The URL used to download the related artifact.
+    :type url: django.db.models.TextField
+
+    Relations:
+
+    :cvar artifact: The artifact that is expected to be present at ``url``.
+    :type artifact: pulp.platform.models.Artifact
+
+    :cvar importer: The importer that contains the configuration necessary
+                    to access ``url``.
+    :type importer: pulp.platform.models.RepositoryImporter
+
+    artifact (file information)
+    importer (download configuration)
+    """
+    url = models.TextField()
+
+    artifact = models.ForeignKey(Artifact, on_delete=models.CASCADE)
+    importer = models.ForeignKey(RepositoryImporter, on_delete=models.CASCADE)
```

Comment:
I *think* making the import non-abstract seems reasonable given this relationship and the likelihood of plans for alt-content sources.